dair_pll/inertia.py

```diff
@@ -203,14 +288,82 @@ def pi_to_theta(pi: Tensor) -> Tensor:
     @staticmethod
-    def pi_to_drake_spatial_inertia(pi: Tensor) -> Tensor:
+    def pi_o_to_pi_cm(pi_o: Tensor) -> Tensor:
+        """Converts batch of ``pi_o`` parameters to ``pi_cm`` parameters using
+        the parallel axis theorem.
+
+        Args:
+            pi_o: ``(*, 10)`` ``pi_o``-type parameterization.
+
+        Returns:
+            pi_cm: ``(*, 10)`` ``pi_cm``-type parameterization.
+        """
+        # Expand in case tensor starts as shape (10,).
+        pi_o = pi_o.reshape(-1, 10)
+
+        # Split ``pi_o`` object into mass, CoM offset, and inertias wrt origin.
+        mass = pi_o[..., 0].reshape(-1, 1)
+        p_BoBcm_B = pi_o[..., 1:4] / mass
+        I_BBo_B = pi_o[..., 4:]
+
+        # Use parallel axis theorem to compute inertia matrix wrt CoM.
+        inertia_mat = inertia_matrix_from_vector(I_BBo_B)
+        new_inertia_mat = parallel_axis_theorem(inertia_mat, mass, p_BoBcm_B,
+                                                from_com_to_other=False)
+        I_BBcm_B = inertia_vector_from_matrix(new_inertia_mat)
+
+        return torch.hstack((mass, p_BoBcm_B*mass, I_BBcm_B)).reshape(-1, 10)
+
+    @staticmethod
+    def pi_cm_to_pi_o(pi_cm: Tensor) -> Tensor:
+        """Converts batch of ``pi_cm`` parameters to ``pi_o`` parameters using
+        the parallel axis theorem.
+
+        Args:
+            pi_cm: ``(*, 10)`` ``pi_cm``-type parameterization.
+
+        Returns:
+            pi_o: ``(*, 10)`` ``pi_o``-type parameterization.
+        """
+        # Expand in case tensor starts as shape (10,).
+        pi_cm = pi_cm.reshape(-1, 10)
+
+        # Split ``pi_cm`` object into mass, CoM offset, and inertias wrt CoM.
+        mass = pi_cm[..., 0].reshape(-1, 1)
+        p_BoBcm_B = pi_cm[..., 1:4] / mass
+        I_BBcm_B = pi_cm[..., 4:]
+
+        # Use parallel axis theorem to compute inertia matrix wrt origin.
+        inertia_mat = inertia_matrix_from_vector(I_BBcm_B)
+        new_inertia_mat = parallel_axis_theorem(inertia_mat, mass, p_BoBcm_B,
+                                                from_com_to_other=True)
+        I_BBo_B = inertia_vector_from_matrix(new_inertia_mat)
+
+        return torch.hstack((mass, p_BoBcm_B*mass, I_BBo_B)).reshape(-1, 10)
+
+    @staticmethod
+    def theta_to_pi_cm(theta: Tensor) -> Tensor:
+        """Passthrough chain of :py:meth:`theta_to_pi_o` and
+        :py:meth:`pi_o_to_pi_cm`."""
+        pi_o = InertialParameterConverter.theta_to_pi_o(theta)
+        return InertialParameterConverter.pi_o_to_pi_cm(pi_o)
+
+    @staticmethod
+    def pi_cm_to_theta(pi_cm: Tensor) -> Tensor:
+        """Passthrough chain of :py:meth:`pi_cm_to_pi_o` and
+        :py:meth:`pi_o_to_theta`."""
+        pi_o = InertialParameterConverter.pi_cm_to_pi_o(pi_cm)
+        return InertialParameterConverter.pi_o_to_theta(pi_o)
+
+    @staticmethod
+    def pi_cm_to_drake_spatial_inertia(pi: Tensor) -> Tensor:
         """Converts batch of ``pi`` parameters to ``drake_spatial_inertia``
```

Comment:
should be `pi_cm` in comment

Reply:
There were a few other instances where `pi` could be distinguished as either `pi_o` or `pi_cm`.  All addressed in [this commit](https://github.com/DAIRLab/dair_pll/pull/4/commits/ae513f85f7ae23ff97ba48a87b46f8d69efee029).